mltsp/util.py

```diff
@@ -139,17 +146,19 @@ def cast_model_params(model_type, model_params):
                 break
         dest_types_list = make_list(param_entry["type"])
         for dest_type in dest_types_list:
-            if dest_type is not str:
-                try:
-                    if isinstance(ast.literal_eval(model_params[k]),
-                                  dest_type):
-                        model_params[k] = ast.literal_eval(model_params[k])
-                        break
-                except ValueError:
-                    pass
-        if isinstance(model_params[k], str) and str not in dest_types_list:
-            raise(ValueError("Model parameter cannot be cast to expected "
-                             "type."))
+            if not all_as_lists:
```

Comment:
How about something like:

```
if not all_as_lists:
    v = [v,]

if all(type(x) in dest_types_list or x is None for x in v):
    break
```
But another question: is this meant to match a specific signature?  Then I guess we can do something like:

```
for actual_type, desired_type in zip(parameters, desired_parameters):
    if not actual_type in desired_type:
        raise ValueError("Model parameters cannot be cast to expected type")
```